### PR TITLE
Add a comparison of two multimedia parameters structs

### PR DESCRIPTION
**Pull request by KobiPesach**
Added the comparison, including a test for it and updated relevant users of mm_np.

### CODE REVIEW

File: liboptv/tests/check_parameters.c [hunk 1]
```diff
@@ -87,11 +118,15 @@ START_TEST(test_read_control_par)
...
+    cpar_correct.mm->d[0] = 52;
+    cpar_correct.mm->n2[1] = 0;
+    cpar_correct.mm->n2[2] = 0;
+    cpar_correct.mm->d[1] = 0;
+    cpar_correct.mm->d[2] = 0;
     
   
     cpar = read_control_par("testing_fodder/parameters/ptv.par");
-    fail_unless(compare_control_par(cpar, &cpar_correct));
+    fail_unless(compare_control_par(cpar, &cpar_correct) == 1);
```
Comment (yosefm) [inline liboptv/tests/check_parameters.c]: I just checked and the documentation only promises a True return value, not necessarily one. So the old line was more correct.

**Comment by yosefm** [comment]
Tested on my system and the current code does not pass the tests. the compare_mm_np test obviously cannot pass, because the two mmp structs are different. In the read_compare_control_par test, one line trips the test. I'll mark it for you with a line comment.

File: liboptv/tests/check_parameters.c [hunk 2]
```diff
@@ -87,11 +118,15 @@ START_TEST(test_read_control_par)
     cpar_correct.mm->n1 = 1;
     cpar_correct.mm->n2[0] = 1.49;
     cpar_correct.mm->n3 = 1.33;
-    cpar_correct.mm->d[0] = 5;
+    cpar_correct.mm->d[0] = 52;
```
Comment (yosefm) [inline liboptv/tests/check_parameters.c]: this change trips the test. It should be 5 as before.

**Comment by yosefm** [comment]
Another thing is, I think we are handling the extra members of d and n2 badly. the comparison function should have a flag instructing it (for now) not to compare those - and so we don't have to set them. It is not quite right to manually set them to 0 when the file-reading function just ignores them.

**Comment by alexlib** [comment]
I tried to test it also, but now I'm stacked with the new policy of cmake:

-- Could NOT find PkgConfig (missing:  PKG_CONFIG_EXECUTABLE) 
-- Found CHECK: /usr/local/lib/libcheck.dylib
CMake Warning (dev) at tests/CMakeLists.txt:25 (get_property):
  Policy CMP0026 is not set: Disallow use of the LOCATION target property.
  Run "cmake --help-policy CMP0026" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The LOCATION property should not be read from target "optv".  Use the
  target name directly with add_custom_command, or use the generator
  expression $<TARGET_FILE>, as appropriate.

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring done
CMake Warning (dev):
  Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
  --help-policy CMP0042" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  MACOSX_RPATH is not specified for the following targets:

   optv

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /Users/alex/Downloads/openptv-1/liboptv

and after that the $ make  fails

**Comment by yosefm** [comment]
This PR didn't change any CMake file. It's either an issue with the previous changes or MacOS. Maybe open a separate issue. I'm writing it down to check on a mac.

**Comment by yosefm** [comment]
Superseded by other commits.

**Pull request closed without merging**